plugins/nf-validation/src/main/nextflow/validation/CustomEvaluators/ExistsEvaluator.groovy

```diff
@@ -0,0 +1,51 @@
+package nextflow.validation
+
+import dev.harrel.jsonschema.Evaluator
+import dev.harrel.jsonschema.EvaluationContext
+import dev.harrel.jsonschema.JsonNode
+import nextflow.Nextflow
+
+import groovy.util.logging.Slf4j
+import java.nio.file.Path
+
+@Slf4j
+class ExistsEvaluator implements Evaluator {
+    // The file should or should not exist
+
+    private final Boolean exists
+
+    ExistsEvaluator(Boolean exists) {
+        this.exists = exists
+    }
+
+    @Override
+    public Evaluator.Result evaluate(EvaluationContext ctx, JsonNode node) {
+        // To stay consistent with other keywords, types not applicable to this keyword should succeed
+        if (!node.isString()) {
+            return Evaluator.Result.success()
+        }
+
+        def String value = node.asString()
+
+        // Skip validation of S3 paths for now
+        if (value.startsWith('s3://')) {
+            log.debug("S3 paths are not supported by 'ExistsEvaluator': '${value}'")
+            return Evaluator.Result.success()
+        }
```

Comment:
Is this consistent with current S3 behaviour or is this a change?

Reply:
This is the same behaviour as before, I'm planning on finding a fix for it though but not in this PR